src/device.c

```diff
@@ -330,31 +338,51 @@ void device_set_thing_name(struct knot_thing *thing, const char *name)
 	strcpy(thing->name, name);
 }
 
+void device_set_thing_ethernet_ip_slave(struct knot_thing *thing,
+					const char *type_plc)
+{
+	strcpy(thing->ethernet_ip_settings.plc_type, type_plc);
+}
+
+void device_set_thing_url(struct knot_thing *thing, const char *url)
+{
+	strcpy(thing->geral_url, url);
+}
+
 void device_set_thing_user_token(struct knot_thing *thing, char *token)
 {
 	thing->user_token = token;
 }
 
-void device_set_thing_modbus_slave(struct knot_thing *thing, int slave_id,
-				   char *url)
+void device_set_thing_modbus_slave(struct knot_thing *thing, int slave_id)
 {
 	thing->modbus_slave.id = slave_id;
-	thing->modbus_slave.url = url;
 }
 
 void device_set_new_data_item(struct knot_thing *thing, int sensor_id,
 			      knot_schema schema, knot_event event,
-			      int reg_addr, int bit_offset, int endianness_type)
+			      int reg_addr, int value_type_size,
+			      int endianness_type, void *driver_buffer)
 {
 	struct knot_data_item *data_item_aux;
+#ifdef DRIVER_ETHERNET_IP
+	struct ethernet_ip_data_settings *ethernet_ip = (struct ethernet_ip_data_settings *) driver_buffer;
+#endif /*DRIVER_ETHERNET_IP*/
 
 	data_item_aux = l_new(struct knot_data_item, 1);
+#ifdef DRIVER_ETHERNET_IP
+	data_item_aux->ethernet_ip_data_settings.element_size = ethernet_ip->element_size;
+	strcpy(data_item_aux->ethernet_ip_data_settings.path,
+	       ethernet_ip->path);
+	strcpy(data_item_aux->ethernet_ip_data_settings.tag_name,
+	       ethernet_ip->tag_name);
+#endif /*DRIVER_ETHERNET_IP*/
```

Comment:
Why not add this driver after setting the data_item_aux values, since move the position on commit 4f4a82c9213e0b49155e6c4b0c428581eb907266